src/store/modules/interactiveMap.ts

```diff
@@ -57,6 +57,7 @@ export interface Card {
   // General simulation fields
   isSimulating: boolean;
   hasSimulationError: boolean;
+  solverStatus: string | null;
```

Comment:
`hasSimulationError` could then just be a computed of `!!solverStatus && solverStatus !== "optimal"`? Maybe not worth all the refactoring.